src/compressed_tensors/offload/load.py

```
@@ -109,25 +122,44 @@ def _get_device_memory() -> dict[int, int]:
     if is_distributed():
-        return {"cpu": _get_shared_memory() - extra_cpu_mem}
+        return {
+            "cpu": _get_shared_memory(num_tensors, total_model_bytes) - extra_cpu_mem
+        }
     else:
         return {"cpu": psutil.virtual_memory().available - extra_cpu_mem}
 
 
-def _get_shared_memory() -> int:
+def _get_shared_memory(num_tensors: int = 0, total_model_bytes: int = 0) -> int:
     linux_shm_path = "/dev/shm"
```

Ruff (0.15.21)

[error] Probable insecure usage of temporary file or directory: "/dev/shm"

(S108)